modules/extensions/src/data-filter/data-filter-extension.ts

```diff
@@ -112,23 +129,29 @@ export default class DataFilterExtension extends LayerExtension<DataFilterExtens
   static extensionName = 'DataFilterExtension';
 
   constructor({
+    categorySize = 1,
     filterSize = 1,
     fp64 = false,
     countItems = false
   }: Partial<DataFilterExtensionOptions> = {}) {
+    if (!DATA_TYPE_FROM_SIZE[categorySize]) {
```

Comment:
Can we allow and just warn?

Reply:
I can't think of a reason why an invalid value should be passed here, as we can't then know how to construct the shader and the app will crash. I've tightened up the types, and will remove these warnings as they should never occur anyway